bidi.py

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/python3
+
+import asyncio
+import json
+import logging
+import socket
+import tempfile
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import aiohttp
+
+log_proto = logging.getLogger("bidi.proto")
+log_command = logging.getLogger("bidi.command")
+
+
+class WebdriverError(RuntimeError):
+    pass
+
+
+class Error(RuntimeError):
+    def __init__(self, msg: str) -> None:
+        self.msg = msg
+
+    def __str__(self) -> str:
+        return self.msg
+
+
+# default timeout
+TIMEOUT = 5  # TODO: raise to 15
+
+
+@dataclass
+class LogMessage:
+    level: str  # like "info"
+    type: str  # usually "console"
+    timestamp: int
+    args: list[object]
+    text: str
+
+    def __init__(self, message_params):
```

Comment:
How do you feel about the cockpit json helpers?  I notice you have your own `JsonObject` defined above, and indeed, we can't import `jsonutil` from `starter-kit`.  I wonder if we should move those helpers to a more neutral location where they can also get used by tests.

It's a bit of a weird situation.  We might end up with two copies... or maybe we don't have to care about typing too much here.

Reply:
Heh yes, had the same "wanna but can't" feeling. But TBH we just need it at exactly one place, I feel it's probably not worth sharing bridge with testlib code for that. Maybe if we run into more of these, but I don't see that coming -- the protocol side is basically "done", the remaining work is all on the "outwards" testlib side.